subsys/bluetooth/host/hci_core.c

```diff
@@ -382,7 +382,7 @@ const bt_addr_le_t *bt_lookup_id_addr(uint8_t id, const bt_addr_le_t *addr)
 #if defined(CONFIG_BT_EXT_ADV)
 uint8_t bt_le_ext_adv_get_index(struct bt_le_ext_adv *adv)
 {
-	uint8_t index = adv - adv_pool;
+	uintptr_t index = adv - adv_pool;
```

Comment:
Can we please add an explicit cast to `uint8_t` in the return statements? Just for good measure :)

Reply:
This is apparently also missing in `bt_le_per_adv_sync_get_index`

Reply:
Yes, added now.